### PR TITLE
refactor: Use schema mapper for appbio design and analysis

### PR DESCRIPTION
**Pull request by nathan-stender**
No change to output data except:

1. Adds "experiment type" to all measurements
2. Fixes a bug in label/unit for melting curve data cube in appbio_quantstudio_designandanalysis_QS7Pro_PCR_with_Melt_example09.json

### CODE REVIEW

File: tests/parsers/appbio_quantstudio_designandanalysis/testdata/appbio_quantstudio_designandanalysis_QS7Pro_PCR_with_Melt_example09.json [hunk 1]
```diff
@@ -501,7 +502,7 @@
                                 }
                             },
                             "melting curve data cube": {
-                                "label": "melt curve",
+                                "label": "melting curve",
```
Comment (nathan-stender) [inline tests/parsers/appbio_quantstudio_designandanalysis/testdata/appbio_quantstudio_designandanalysis_QS7Pro_PCR_with_Melt_example09.json:505]: NOTE to reviewer: this and the unit change in were bugs in the existing implementation

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py [hunk 2]
```diff
@@ -32,66 +32,66 @@
...
+        ),
+    ]
+
+
+def _create_processed_data(
+    amplification_data: AmplificationData, result: Result
+) -> ProcessedData:
     return ProcessedData(
-        automatic_cycle_threshold_enabled_setting=result.automatic_baseline_determination_enabled_setting,
+        automatic_cycle_threshold_enabled_setting=result.automatic_cycle_threshold_enabled_setting,
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_data_creator.py:73]: NOTE to reviewer: This was a bug in the existing version

File: src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_structure.py [hunk 3]
```diff
@@ -248,30 +248,32 @@ class AmplificationData:
...
-        assert_not_none(
-            reader.drop_until(r"^\[Amplification Data\]"),
-            msg="Unable to find 'Amplification Data' section in file.",
-        )
 
-        reader.pop()  # remove title
-        lines = list(reader.pop_until(r"^\[.+\]"))
-        csv_stream = StringIO("\n".join(lines))
-        data = read_csv(csv_stream, sep="\t", thousands=r",")
+def create_amplification_data(
```
Comment (nathan-stender) [inline src/allotropy/parsers/appbio_quantstudio/appbio_quantstudio_structure.py:252]: Note to reviewer: this allows re-use of `AmplificationData` in design and analysis parser

File: tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_example05.json [hunk 4]
```diff
@@ -540,7 +540,7 @@
                                                 "value": 0.06,
                                                 "unit": "(unitless)"
                                             },
-                                            "automatic cycle threshold enabled setting": true,
+                                            "automatic cycle threshold enabled setting": false,
```
Comment (nathan-stender) [inline tests/parsers/appbio_quantstudio/testdata/appbio_quantstudio_example05.json:543]: NOTE to reviewer: these changes are due to a bug in the existing version, fixed here.

**Review by stephenworlow — DISMISSED** [review]

**Review by stephenworlow — APPROVED** [review]

**Pull request merged**